libminifi/include/c2/C2Agent.h

```diff
@@ -244,6 +239,10 @@ class C2Agent : public state::UpdateController, public state::response::Response
   std::string bin_location_;
 
   std::shared_ptr<logging::Logger> logger_;
+
+  utils::ThreadPool<state::Update> heartbeat_thread_pool_;
```

Comment:
I'm afraid it's going to conflict with my threadpool PR, which is about having one single threadpool.
In case that gets merged before this, let's have a talk to fit this in that architecture.